source/standalone/workflows/skrl/train.py

```diff
@@ -0,0 +1,223 @@
+    # - False: the interaction with the environment will be performed manually.
+    #          This mode allows recording specific information about the environment
+
+    # configure and instantiate the RL trainer
+    # https://skrl.readthedocs.io/en/latest/modules/skrl.trainers.sequential.html
+    if experiment_cfg["trainer"]["use_api"]:
+        trainer_cfg = experiment_cfg["trainer"]
+        trainer = SequentialTrainer(cfg=trainer_cfg, env=env, agents=agent)
+
+        # train the agent
+        trainer.train()
+
+        # close the simulator (the environment is automatically closed by skrl)
+        simulation_app.close()
+
+    # execute the interaction with the environment without using a trainer
+    # (note that skrl requires the execution of some additional agent methods for proper operation).
+    # https://skrl.readthedocs.io/en/latest/modules/skrl.trainers.manual.html
+    else:
+        import tqdm
+
+        timesteps = experiment_cfg["trainer"]["timesteps"]
+        agent.init()
+        agent.set_running_mode("train")
+
+        # reset environment
+        obs, infos = env.reset()
+        # simulate environment
+        for timestep in tqdm.tqdm(range(timesteps)):
+            timestep += 1
+            # pre-interaction stepping
+            agent.pre_interaction(timestep=timestep, timesteps=timesteps)
+            # agent stepping
+            with torch.no_grad():
+                actions = agent.act(obs, timestep=timestep, timesteps=timesteps)[0]
+            # env stepping
+            next_obs, rewards, terminated, truncated, infos = env.step(actions)
+            # record environment transitions and track data
+            with torch.no_grad():
+                agent.record_transition(
+                    states=obs,
+                    actions=actions,
+                    rewards=rewards,
+                    next_states=next_obs,
+                    terminated=terminated,
+                    truncated=truncated,
+                    infos=infos,
+                    timestep=timestep,
+                    timesteps=timesteps,
+                )
+            # log custom environment data
+            if "episode" in infos:
+                for k, v in infos["episode"].items():
+                    if isinstance(v, torch.Tensor) and v.numel() == 1:
+                        agent.track_data(f"Info / {k}", v.item())
+            # post-interaction stepping (write data to TensorBoard / Weights & Biases)
+            agent.post_interaction(timestep=timestep, timesteps=timesteps)
+            # reset environments
+            with torch.no_grad():
+                if terminated.any() or truncated.any():
+                    obs, infos = env.reset()
+                else:
+                    obs.copy_(next_obs)
+            # check if simulator is stopped
+            if env.sim.is_stopped():
+                break
+
+        # close the simulator
+        env.close()
+        simulation_app.close()
```

Comment:
Do we need this more custom version of the trainer? For training, it is fine to use a trainer, because we assume the user would handle their custom trainers from the library side and not over here. 

I feel right now it adds some bulk and makes the script longer than it needs to be. Maybe I am missing something here.

Reply:
Well, running all those steps to train/evaluate the agent is the only way to track custom data (Episode separated rewards)...
Because the `''episode"` key in the `info` dictionary is not an standard, library code does not track It by default

Reply:
```python
            # log custom environment data
            if "episode" in infos:
                for k, v in infos["episode"].items():
                    if isinstance(v, torch.Tensor) and v.numel() == 1:
                        agent.track_data(f"Info / {k}", v.item())
```